metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -485,6 +491,32 @@ def _get_dataset_column_sample_values(
                 str(v) for v in res["partial_unexpected_list"]
             ]
 
+    def run_stage_3_optimized_queries(
+        self, columns_to_profile: List[_SingleColumnSpec]
+    ) -> None:
+        if self.config.include_field_sample_values and isinstance(
+            self.dataset, SqlAlchemyDataset
+        ):
+            sample_values_query = (
+                sa.select(
+                    [
+                        sa.column(column_spec.column)
+                        for column_spec in columns_to_profile
+                    ]
+                )
+                .limit(SAMPLE_VALUES_LIMIT)
+                .select_from(self.dataset._table)
+            )
+
+            sample_values = self.dataset.engine.execute(sample_values_query).fetchall()
```

Comment:
There is a slight difference between GE and your implementation, and in yours, it can happen if the 20 sample value for a column will be null, then we don't get any sample for that column. While GE's query only get the non-null values for every column.
I think this is something we can live with but worth noting.

GE runs the following query:
`SELECT pricepaid FROM public.sales WHERE 1 = 1 AND pricepaid IS NOT NULL LIMIT 20`

Reply:
Ah, you're right @treff7es. Thanks for calling this out!